src/pages/JoinStep2/JoinStep2.js

```diff
@@ -0,0 +1,365 @@
+    this.setState({
+      [name]: String(value)
+    });
+    console.log(this.state);
+  };
+
+  handleSMSCheckbox = e => {
+    const { name } = e.target;
+    this.setState({
+      [name]: !this.state.smsAgreement
+    });
+    console.log(this.state);
+  };
+
+  handleEmailCheckbox = e => {
+    const { name } = e.target;
+    this.setState({
+      [name]: !this.state.emailAgreement
+    });
+    console.log(this.state);
+  };
+
+  checkId = () => {
+    const APIforIdCheck = "http://localhost:3000/data/account.json";
+    const { account } = this.state;
+    fetch(APIforIdCheck, {
+      method: "POST",
+      body: JSON.stringify({
+        account: account
+      })
+    })
+      .then(response => response.json())
+      .then(result => {
+        console.log("백엔드에서 오는 응답 메세지:", result);
+      });
+  };
+
+  checkEmail = () => {
+    const APIforEmailCheck = "http://localhost:3000/data/email.json";
+    const { email } = this.state;
+    fetch(APIforEmailCheck, {
+      method: "POST",
+      body: JSON.stringify({
+        email: email
+      })
+    })
+      .then(response => response.json())
+      .then(result => {
+        console.log("백엔드에서 오는 응답 메세지:", result);
+      });
+  };
+
+  isPwInfoValid = password => {
+    let number = /[0-9]/g;
+    let character = /[a-zA-z]/g;
+    let specialCharacter = /\W|_/g;
+    const isNumIncluded = password.match(number) ? true : false;
+    const isCharIncluded = password.match(character) ? true : false;
+    const isSpecialCharIncluded = password.match(specialCharacter)
+      ? true
+      : false;
+    const isPwLong = password.length >= 6 && password.length <= 16;
+    if (
+      ((isNumIncluded && isSpecialCharIncluded) ||
+        (isNumIncluded && isCharIncluded) ||
+        (isCharIncluded && isSpecialCharIncluded)) &&
+      isPwLong
+    ) {
+      return true;
+    } else return false;
+  };
+
+  isInputNull = () => {
+    const { name, account, password, email, phoneNumber } = this.state;
+    name === ""
+      ? this.setState({ isNameNull: false })
+      : this.setState({ isNameNull: true });
+    account === ""
+      ? this.setState({ isAccountNull: false })
+      : this.setState({ isAccountNull: true });
+    password === ""
+      ? this.setState({ isPasswordNull: false })
+      : this.setState({ isPasswordNull: true });
+    email === ""
+      ? this.setState({ isEmailNull: false })
+      : this.setState({ isEmailNull: true });
+    phoneNumber === ""
+      ? this.setState({ isPhoneNumberNull: false })
+      : this.setState({ isPhoneNumberNull: true });
+  };
+
+  checkValidation = e => {
+    e.preventDefault();
+    const {
+      name,
+      account,
+      password,
+      confirmPw,
+      email,
+      phoneNumber,
+      dateOfBirth,
+      smsAgreement,
+      emailAgreement
+    } = this.state;
+    const isEssentialInfoValid =
+      name !== "" &&
+      account !== "" &&
+      password !== "" &&
+      email !== "" &&
+      phoneNumber !== "";
```

Comment:
```suggestion
const isEssentialInfoValid =
      !!name && !!account && !!password && !!email && !!phoneNumber;
```
"false가 아닐 때"라는 말은 "true일 때"라는 말과 같습니다